### ISSUE TITLE
Syntax highlighting via Shiki requires additional configuration

### ISSUE THREAD

**Issue by adamdehaven** [nuxt-modules/mdc#159]
@farnabaz I'm working on implementing the `<MDCRenderer>` via our changes in #154 and am running into an issue whereby I cannot get wasm to work properly on Cloudflare Pages.

Small reproduction repo: https://github.com/adamdehaven/mdc-wasm-repro

~~I enabled experimental `wasm` support in my `nuxt.config.ts` `nitro.experimental.wasm: true` (I think this is needed?) but then when using `wrangler` dev preview chokes.~~

I _think_ I need to somehow include the `shiki/onig.wasm` locally in my project based on the [Shiki docs](https://shiki.style/guide/install#cloudflare-workers), but unsure how/where to do so. 

![Shiki Cloudflare Workers and wasm](https://github.com/nuxt-modules/mdc/assets/2229946/d50c6fe1-3b1f-4393-ad8c-4fd15560e766)

Would you be able to provide some guidance? The markdown itself renders; however, syntax highlighting obviously is not currently working for me.

**Comment by farnabaz** [comment]
You don't need to enable Nitro's experimental WASM. MDC module should handle WASM configuration for you. Just remove the experimental option and try the preview.

**Comment by adamdehaven** [comment]
When I simply use the MDC module in a regular Nuxt project, I get a different error:

> WebAssembly.instantiate(): Wasm code generation disallowed by embedder

**Comment by adamdehaven** [comment]
@farnabaz I've put together a small reproduction: https://github.com/adamdehaven/mdc-wasm-repro with simple instructions.

Install deps, and then if you run `pnpm dev` or even `pnpm build && pnpm preview` everything locally works fine. 

However, if you run `pnpm build` and then `pnpm wrangler pages dev dist` to serve with wrangler, you can successfully load the app and the markdown renders, but the syntax highlighting is not functional and you see the following wasm error in your console:

![MDC wasm error](https://github.com/nuxt-modules/mdc/assets/2229946/c8d449c0-5554-46d7-91f8-5d8fddd092ef)

This matches what I see when I do an actual deploy to Cloudflare pages.

**Comment by farnabaz** [comment]
I did manage to run and also deploy your reproduction is first attempt. https://a7626d8f.mdc-test-a.pages.dev/
Also, I did manage to deploy my fork directly from Cloudflare dashboard https://mdc-wasm-repro.pages.dev/

Something might be happening in your machine that cause this behavior.
- Check your node version (try with 18)
- Try force install maybe `pnpm i --force`

**Comment by adamdehaven** [comment]
@farnabaz if you look at either of your deploy links above, notice the syntax highlighting is not working. The deploy itself works, but the Shiki syntax highlighting isn't functional.

## pnpm run dev (local)

(notice the colors in the code block)

![image](https://github.com/nuxt-modules/mdc/assets/2229946/8fb8609f-27e0-4e30-ba53-cfa1ad25c6f0)

## Cloudflare Pages deploy 

(notice no syntax highlighting like on local)

![image](https://github.com/nuxt-modules/mdc/assets/2229946/029efde1-ca80-4250-b0ba-fcf22a7a3f8f)

**Comment by farnabaz** [comment]
My bad, I thought it was raising error on deployment.
I'll dig this

**Comment by adamdehaven** [comment]
@farnabaz so I saw the [Shiki Nuxt blog post published today](https://nuxt.com/blog/shiki-v1) which indicates this should work out of the box with Cloudflare workers/Pages. 

What are we missing? 🤔 I dug in over the weekend but unfortunately didn't find anything that made this work

**Comment by adamdehaven** [comment]
Ok I think I figured it out with help from @Atinux 🎉 but may need help figuring out the best place to supplement the documentation.

Turns out it is undocumented (I believe) that you need to add `shiki/onig.wasm` in your `nuxt.config` > `vite.build.rollupOptions.external` array 😅 in combination with enabling `nitro.experimental.wasm`.

Adding these two settings fixes syntax highlighting for the Cloudflare Pages deploy ✨ however, it breaks syntax highlighting for `dev` mode.

I think the easiest solution would be to only enable the experimental wasm for production, meaning your `nuxt.config` would look similar to this: 

```js
$production: {
  nitro: {
    // !Important: we only want to enable the wasm feature in production since it will break syntax highlighting when running the dev server
    experimental: {
      wasm: true,
    },
  },
},
vite: {
  build: {
    rollupOptions: {
      external: [
        'shiki/onig.wasm', // !Important: externalize the wasm import
      ],
    },
  },
},
```

Is adding the experimental wasm setting only for `$production` ideal here, or is there a better optimization that would be ideal?

**Comment by adamdehaven** [comment]
Here's a PR to supplement the documentation: https://github.com/nuxt-modules/mdc/pull/165

Ideally, I'd like to see if it's possible to leave the `wasm` enabled for `$development` as well

**Comment by farnabaz** [comment]
Should be fixed in #167
I just deployed your repro with `@nuxtjs/mdc-edge@latest` https://45865333.mdc-wasm-repro.pages.dev/
